packages/lib-sourcify/src/Validation/zipUtils.ts

```diff
@@ -0,0 +1,74 @@
+import JSZip from 'jszip';
+import { logDebug } from '../lib/logger';
+import { PathBuffer } from '../lib/types';
+
+export async function unzipFiles(files: PathBuffer[]) {
+  const allUnzipped: PathBuffer[] = [];
+  for (let i = 0; i < files.length; i++) {
+    const file = files[i];
+    if (isZip(file.buffer)) {
+      logDebug('Unzipping file', {
+        path: file.path,
+        size: file.buffer.length,
+        unzippedFilesCount: allUnzipped.length,
+      });
+      const unzipped = await unzip(file);
+      allUnzipped.push(...unzipped);
+
+      logDebug('Unzipped file', {
+        path: file.path,
+        size: file.buffer.length,
+        unzippedFilesCount: allUnzipped.length,
+      });
+      // Remove the zip file from the array and decrement the index to check the next file.
+      files.splice(i, 1);
+      i--;
+    }
+  }
+  // Add unzipped at the end to not check again if the extracted files are zips.
+  files.push(...allUnzipped);
+}
+
+function isZip(file: Buffer): boolean {
```

Comment:
Same comment as `fetchUtils.ts`. I don't like them in the same folder as `SolidityMetadataContract.ts` and `processFiles.ts`. What about creating a `utils/` folder in which we put both this and `fetchUtils.ts`?

Reply:
As said, I don't see it being used anywhere else, so it's actually only validation related utils. If that's shared with some other parts of the code, then yes. For me this structure is quite clear.